### PR TITLE
kops: Pick up right kubectl / bump e2e-image

### PR DESCRIPTION
**Pull request by zmerlynn**
First part of fix for https://github.com/kubernetes/kops/issues/1278

### CODE REVIEW

File: jenkins/e2e-image/e2e-runner.sh [hunk 1]
```diff
@@ -432,6 +432,12 @@ elif [[ ! "${JOB_NAME}" =~ -pull- ]]; then
   fi
 fi
 
+if [[ -n "${PRIORITY_PATH}" ]]; then
```
Comment (spxtr) [inline jenkins/e2e-image/e2e-runner.sh]: Should this be `${PRIORITY_PATH:-}`?

Comment (zmerlynn) [inline jenkins/e2e-image/e2e-runner.sh]: Yes

Comment (zmerlynn) [inline jenkins/e2e-image/e2e-runner.sh]: Fixed, re-pushed.

**Comment by zmerlynn** [comment]
Testing in https://github.com/kubernetes/kubernetes/pull/39253

File: jenkins/e2e-image/e2e-runner.sh [hunk 2]
```diff
@@ -432,6 +432,12 @@ elif [[ ! "${JOB_NAME}" =~ -pull- ]]; then
   fi
 fi
 
+if [[ -n "${PRIORITY_PATH:-}" ]]; then
+  export PATH="${PRIORITY_PATH}:${PATH}"
+fi
+echo "kubectl on path is:"
+kubectl version --client
```
Comment (spxtr) [inline jenkins/e2e-image/e2e-runner.sh]: I think we print out the version in `e2e.go`. Should we also do it here?

Comment (zmerlynn) [inline jenkins/e2e-image/e2e-runner.sh]: It's something we print if the cluster comes up (and we collect the server version then), but this is recording it prior to that. I can omit it if you'd like, or promise to come back around and clean it up in e2e.go.

Comment (spxtr) [inline jenkins/e2e-image/e2e-runner.sh]: I'd rather we add `--print_client_version` to `e2e.go` or something, but I don't think it's a very big deal. @ixdy has been working on `kubectl version` I think, might want to weigh in.

Comment (zmerlynn) [inline jenkins/e2e-image/e2e-runner.sh]: I'm actually going to take it out. Something is too subtle here and it broke all of the *other* PR builders when I tried this, and I'd rather just get the other thing fixed.

Comment (zmerlynn) [inline jenkins/e2e-image/e2e-runner.sh]: PTAL

**Comment by zmerlynn** [comment]
Will wait for PR builder to finish and merge.

File: jenkins/e2e-image/e2e-runner.sh (hunk 2 above)
Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: @spxtr you're probably thinking of https://github.com/kubernetes/kubernetes/pull/37049, which is only tangentially related, needs work, and is unclear when I'll have time to look at again.

Comment (spxtr) [inline jenkins/e2e-image/e2e-runner.sh]: That's the one.

**Pull request merged**